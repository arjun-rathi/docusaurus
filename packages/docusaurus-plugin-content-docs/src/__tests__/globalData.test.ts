/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {toGlobalDataVersion} from '../globalData';

describe('toGlobalDataVersion', () => {
  it('generates the right docs, sidebars, and metadata', () => {
    expect(
      toGlobalDataVersion({
        versionName: 'current',
        versionLabel: 'Label',
        isLast: true,
        versionPath: '/current',
        mainDocId: 'main',
        docs: [
          {
            unversionedId: 'main',
            permalink: '/current/main',
            sidebar: 'tutorial',
          },
          {
            unversionedId: 'doc',
            permalink: '/current/doc',
            sidebar: 'tutorial',
          },
        ],
        sidebars: {
          another: [
            {
              type: 'category',
              label: 'Generated',
              link: {
                type: 'generated-index',
                permalink: '/current/generated',
              },
              items: [
                {
                  type: 'doc',
                  id: 'doc',
                },
              ],
            },
          ],
          tutorial: [
            {
              type: 'doc',
              id: 'main',
            },
            {
              type: 'category',
              label: 'Generated',
              link: {
                type: 'generated-index',
                permalink: '/current/generated',
              },
              items: [
                {
                  type: 'doc',
                  id: 'doc',
                },
              ],
            },
          ],
          links: [
            {
              type: 'link',
              href: 'foo',
              label: 'Foo',
            },
            {
              type: 'link',
              href: 'bar',
              label: 'Bar',
            },
          ],
        },
        categoryGeneratedIndices: [
          {
            title: 'Generated',
            slug: '/current/generated',
            permalink: '/current/generated',
            sidebar: 'tutorial',
          },
        ],
        versionBanner: 'unreleased',
        versionBadge: true,
        versionClassName: 'current-cls',
        tagsPath: '/current/tags',
        contentPath: '',
        contentPathLocalized: '',
        sidebarFilePath: '',
        routePriority: 0.5,
      }),
    ).toMatchSnapshot();
  });
});
