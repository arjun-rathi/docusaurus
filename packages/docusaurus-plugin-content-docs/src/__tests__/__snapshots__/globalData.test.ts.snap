// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toGlobalDataVersion generates the right docs, sidebars, and metadata 1`] = `
{
  "docs": [
    {
      "id": "main",
      "path": "/current/main",
      "sidebar": "tutorial",
    },
    {
      "id": "doc",
      "path": "/current/doc",
      "sidebar": "tutorial",
    },
    {
      "id": "/current/generated",
      "path": "/current/generated",
      "sidebar": "tutorial",
    },
  ],
  "isLast": true,
  "label": "Label",
  "mainDocId": "main",
  "name": "current",
  "path": "/current",
  "sidebars": {
    "another": {
      "link": {
        "label": "Generated",
        "path": "/current/generated",
      },
    },
    "links": {},
    "tutorial": {
      "link": {
        "label": "main",
        "path": "/current/main",
      },
    },
  },
}
`;
